runtime/doc/options.txt

```diff
@@ -2186,6 +2186,26 @@ A jump table for the options with a short description can be found at |Q_op|.
 	This option is reset when the 'paste' option is set and restored when
 	the 'paste' option is reset.
 
+				*'filechangenotify'* *'fcnotify'*
+'filechangenotify' 'fcnotify'	string (default "off")
```

Comment:
Why are we adding an option for this? If users want/need to deal with these low-level details beyond simply setting 'autoread', that can be done by adding/deleting autocmds. I don't see why we would want to crystallize this in a new option (which is largely redundant with 'autoread').

Reply:
One of the reasons were that we might want to support different backends for watching the files. (Some backends support remote filesystems, while libuv doesn't.)

Also, we can either have a libuv based watcher for watching the files, or we could check for the changes on `FocusGained` (which came from: https://github.com/neovim/neovim/pull/12531). There's no point in having both of them active at the same time, and I don't think it is possible to verify if the watcher is working, so that we can turn the autoread tui pr "off". We can not decide what we should use if we have autoread set. Since the watcher might not work on some filesystems, (like remote filesystems,) setting autoread on those systems would render the option useless. It would be better if we could simply use your autoread tui PR for remote filesystems.

Autoread was, before the autoread-tui PR was merged, an option that dictated the behavior of `checktime`. I don't think we ever automatically reloaded a file just because it was changed outside neovim. That is what the issue was all about. 

What I feel is, autoread currently is supposed to do two separate things:
1. Checking for if a file was changed outside of neovim.
2. In case the file was changed, seeing if we want to notify the user or if we want to reload the buffer.

What autoread has been doing initially is only number 2, although the documentation says it will be doing both 1 & 2, which is where the confusion and false expectations from autoread came from. I think we need to have 2 separate options for these since these are 2 separate things.

A user might not want to watch a file using the fs_watch handle at all, but might still want to reload the buffer everytime `:checktime` is executed, because of old configs. Or a user might want to watch a file using `fs_watch`, but might not want it to be reloaded automatically. He would like to be notified for the changes so that he could decide if he wants the file to be reloaded or not. Both of these things can not be controlled by autoread alone.

Reply:
I think I have an idea. I would have to discuss it with others first though.

Reply:
see https://github.com/neovim/neovim/issues/12605#issuecomment-1441796934